### PR TITLE
Changelog v.2.0.0

### PR DESCRIPTION
**Pull request by BridgeAR**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,10 +1,72 @@
...
-* [#815](https://github.com/NodeRedis/node_redis/pull/815) Consistently use new debug functionality (@BridgeAR)
-* [#814](https://github.com/NodeRedis/node_redis/pull/814) Support new commands and drop support for deprecated 'substr' (@BridgeAR)
+## v2.0.0 - Sep 21, 2015
+
+This is the biggest release that node_redis had since it was released in 2010. A long list of outstanding bugs has been fixed and we are very happy to present you redis 2.0 and we highly recommend updating as soon as possible.
+
+#What's new in 2.0
+
+- Implemented a "connection is broken" mode if no connection could be established
+- node_redis should not take any apps down anymore by behaving weird
```
Comment (bcoe) [inline changelog.md]: might change this to something like:

> node_redis no longer throws under any circumstances, preventing it from terminating applications.

File: changelog.md [hunk 2]
```diff
@@ -1,10 +1,72 @@
...
+
+- Added a "redis connection is broken" mode after reaching max connection attempts / exceeding connection timeout. (@BridgeAR)
+- Added NODE_DEBUG=redis env to activate the debug_mode (@BridgeAR)
+- Added a default connection timeout of 24h instead of never timing out as a default (@BridgeAR)
+- Errors thrown by redis will now include the redis error code as ".code" property. (@skeggse & @BridgeAR)
+- Errors thrown by node_redis will now include a ".command" property for the command used (@BridgeAR)
+- The entries in the keyspace of the server_info is now an object instead of a string. (@SinisterLight & @BridgeAR)
+- Small speedup here and there (e.g. by not using .toLowerCase() anymore) (@BridgeAR)
+- Full windows support (@bcoe)
+- Increased coverage by 10% and add a lot of tests to make sure everything works as it should (We reached 97% now) (@BridgeAR)
```
Comment (bcoe) [inline changelog.md]: \o/ amazing work @BridgeAR

File: changelog.md [hunk 3]
```diff
@@ -1,10 +1,72 @@
...
+- Fix a couple small other bugs
+
+## Breaking changes:
+
+ ... connection is broken mode ...
+ ... multi error handling ...
+ ... not catching user errors anymore ...
+ ... emitting errors instead of throwing them sync ...
+ ... keyspace of the server_info
+ ... send_command commands have to be lowerCase
```
Comment (bcoe) [inline changelog.md]: might it be worth having a `.toLowerCase()` to protect legacy users, or would you rather just bite the bullet and force people to lower case commands?

File: README.md [hunk 4]
```diff
@@ -617,110 +616,53 @@ Much effort has been spent to make `node_redis` as fast as possible for common
...
-
-If you use very large responses in your application, the JavaScript parser
-performs badly. Until the JS parser is fixed, you can use the C-based `hiredis`
-parser bound to the official `hiredis` C library. To use `hiredis`, do:
+Here are results of `multi_bench.js` which is similar to `redis-benchmark` from the Redis distribution.
+
+hiredis parser (Lenovo T450s i7-5600U):
+
+    Client count: 5, node version: 2.5.0, server version: 3.0.3, parser: hiredis
+             PING,     1/5 min/max/avg/p95:    0/   3/   0.03/   0.00    540ms total, 37037.04 ops/sec
```
Comment (bcoe) [inline README.md]: these benchmarks rock!

**Comment by bcoe** [comment]
@BridgeAR it would be good to link the changelog to issues where possible, I find this is helpful. Otherwise, this is a pretty amazing amount of work:

* the 97% test coverage makes me happy.
* the benchmarks are promising.
* it's great that so many tickets were cleaned up.

We should also loop in: @brycebaril, @erinspice, and @blainsmith

File: changelog.md (hunk 3 above)
Comment (BridgeAR) [inline changelog.md]: I'd like them to bite the bullet. This is only an issue for people who use send_command directly and I guess that most people use redis.command instead. And that is still usable as upper case command.

**Pull request merged**